Types.d.ts

```diff
@@ -0,0 +1,36 @@
+export type GenericMeta =
+  {
+    // ???
+  };
+
+export type BareHeaders = Map<string, string | string[]>;
+
+export type SmallResponse =
+  {
+    body: ReadableStream | ArrayBuffer | Blob | string,
+    headers: BareHeaders,
+    status: number
+  }
+
+export interface BareTransport {
+  connect: (
+    url: URL,
+    protocols: [string],
+    onopen: () => void,
+    onmessage: (data: ArrayBuffer | string) => void,
```

Comment:
```suggestion
    onmessage: (data: Blob | string) => void,
```
Blobs can be made into ArrayBuffers (and vice versa) with relative ease, but they're also better for large messages. Plus, WSes themselves have [an option](https://developer.mozilla.org/en-US/docs/Web/API/WebSocket/binaryType) to use Blobs instead of ArrayBuffers, so this benefit isn't something I'm making up.

Reply:
Actually, I've changed my mind. ArrayBuffers might make more sense, as constructing a Blob requires all the parts to be present in memory anyway.